reader/src/main/java/org/jline/reader/impl/LineReaderImpl.java

```diff
@@ -781,15 +788,15 @@ public void printAbove(String str) {
             if (reading) {
                 display.update(Collections.emptyList(), 0);
             }
-            if (str.endsWith("\n") || str.endsWith("\n\033[m") || str.endsWith("\n\033[0m")) {
+            if (str.endsWith(LF) || str.endsWith("\n\033[m") || str.endsWith("\n\033[0m")) {
```

Comment:
I would leave that change out.  The `CR` and `LF` constants were meant to make it more obvious when dealing with those cr/lf issues.